### DISCUSSION TITLE
How to make collision with our own obj model

### DISCUSSION THREAD

**Discussion by ghost** [InteractiveComputerGraphics/PositionBasedDynamics#147] [Q&A] [answered]
Hi, 

Could you please provide an example on how to make collision with our own obj model ? from the file "ClothCollisionDemo.cpp" I tried to replace the torus obj file with the "armadillo" obj file but the cloth still collide with the torus object shape even tough it is not loaded.

Thank you.

**Comment by janbender** [comment]
The problem is that in line 266 a torus collisison object is created: 

https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/blob/master/Demos/DistanceFieldDemos/ClothCollisionDemo.cpp#L266

Instead of cd.addCollisionTorus() you should use cd.addCubicSDFCollisionObject()

Hope that helps.

**Reply by ghost** [reply]
Thank you for the info, I am trying to understand how to fill the function argument "GridPtr sdf". I just need to spend some time on the demo of scene loader it looks like it uses the function.

**Comment by janbender** [comment, marked as answer]
Try with this code... 

[DistanceFieldDemos.zip](https://github.com/InteractiveComputerGraphics/PositionBasedDynamics/files/10087162/DistanceFieldDemos.zip)

**Reply by ghost** [reply]
Thank you very much.
